### DISCUSSION TITLE
HTTP header authentication not working with gradle publish

### DISCUSSION THREAD

**Discussion by igor-sales-newday** [community/community#27956] [Actions] [answered]
**The following documentation describes how to use HTTP header authentication:**
https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:authentication_schemes

**I did as described, answer: Received status code 401 from server: Unauthorized**

    publishing {
    	repositories {
    		maven {
    			url "my-url-does-not-matter-now"
    			
    			credentials(HttpHeaderCredentials) {
    				name = "Private-Token"
    				value = "my-token"
    			}
    			authentication {
    				header(HttpHeaderAuthentication)
    			}
    		}
    	}
    }

**Interesting enough, if i change to password protected works.**

    publishing {
    	repositories {
    		maven {
    			url "my-url-does-not-matter-now"
    			credentials {
    				username "user"
    				password "my-token"
    			}
    		}
    	}
    }

I can assume my url and token works.

Thoughts please? thanks in advance

**Comment by airtower-luna** [comment, marked as answer]
I don't have experience with Gradle specifically, but from a glance at those docs it looks like that config will add a head `Private-Token` with the value `my-token`. In all other places I've seen and used, GitHub requires an `Authorization` header with the value `token YOUR_PAT_HERE`, as described here: https://docs.github.com/en/rest/overview/resources-in-the-rest-api#oauth2-token-sent-in-a-header

**Reply by igor-sales-newday** [reply]
Thanks for sharing that, yesterday night i came home and had similar idea,
I just replaced name = "Private-Token" with "Authorization" and did work.
